src/main/java/io/odpf/firehose/sink/objectstorage/writer/remote/ObjectStorageChecker.java

```diff
@@ -13,40 +20,97 @@
 import java.util.concurrent.ExecutorService;
 import java.util.stream.Collectors;
 
+import static io.odpf.firehose.metrics.Metrics.*;
+import static io.odpf.firehose.sink.objectstorage.ObjectStorageMetrics.*;
+
 @AllArgsConstructor
 public class ObjectStorageChecker implements Runnable {
     private static final Logger LOGGER = LoggerFactory.getLogger(ObjectStorageChecker.class);
-    private final BlockingQueue<String> toBeFlushedToRemotePaths;
+
+    private final BlockingQueue<FileMeta> toBeFlushedToRemotePaths;
     private final BlockingQueue<String> flushedToRemotePaths;
     private final Set<ObjectStorageWriterWorkerFuture> remoteUploadFutures;
     private final ExecutorService remoteUploadScheduler;
     private final ObjectStorage objectStorage;
+    private final Clock clock;
+    private final Instrumentation instrumentation;
 
     @Override
     public void run() {
-        List<String> tobeFlushed = new ArrayList<>();
+        List<FileMeta> tobeFlushed = new ArrayList<>();
         toBeFlushedToRemotePaths.drainTo(tobeFlushed);
         remoteUploadFutures.addAll(tobeFlushed.stream()
-                .map(path -> new ObjectStorageWriterWorkerFuture(
-                        remoteUploadScheduler.submit(() -> objectStorage.store(path)), path)
+                .map(fileMeta -> new ObjectStorageWriterWorkerFuture(
+                        remoteUploadScheduler.submit(() -> {
+                            try {
+                                objectStorage.store(fileMeta.getFullPath());
+                            } catch (Exception e) {
+                                throw new RuntimeException(e);
+                            }
+                        }), fileMeta, clock.now())
                 ).collect(Collectors.toList()));
 
-        Set<String> flushedPath = remoteUploadFutures.stream().map(future -> {
-            if (!future.getFuture().isDone()) {
+        Set<String> flushedPath = remoteUploadFutures.stream().map(uploadJob -> {
+            if (!uploadJob.getFuture().isDone()) {
                 return "";
             } else {
                 try {
-                    future.getFuture().get();
-                    LOGGER.info("Flushed to Object storage " + future.getPath());
-                } catch (InterruptedException e) {
-                    throw new ObjectStorageFailedException(e);
-                } catch (ExecutionException e) {
-                    throw new ObjectStorageFailedException(e.getCause());
+                    uploadJob.getFuture().get();
+                    FileMeta fileMeta = uploadJob.getFileMeta();
+                    Partition partition = fileMeta.getPartition();
+
+                    LOGGER.info("Flushed to Object storage " + fileMeta.getFullPath());
+
+                    String topic = partition.getTopic();
+                    String datetimeTag = partition.getDatetimePathWithoutPrefix();
+
+                    instrumentation.incrementCounterWithTags(FILE_UPLOAD_TOTAL,
```

Comment:
can we put all this instrumentation in a separate method.. it's very messy here.

Reply:
call a private method with filemeta (or anything else is needed ) as argument.